lib/spdx.rb

```diff
@@ -178,4 +179,23 @@ def self.special_cases # rubocop:disable Metrics/MethodLength
   def self.names
     (SpdxLicenses.data.keys + SpdxLicenses.data.map { |_k, v| v['name'] }).sort
   end
+
+  def self.valid_spdx?(spdx_string)
+    eval(Parser.parse_to_ruby(spdx_string))
+    return true
+  rescue SyntaxError, Exception
+    return false
+  end
+
+  def self.can_use_with_allow?(spdx_string, allowed_licenses)
+    licenses = allowed_licenses
+    eval(Parser.parse_to_ruby(spdx_string))
```

Comment:
This is really cool, but my spidey sense tells me we're opening up an attack vector. Let's chat about it later toady.

Reply:
Yeah, I'm super wary of `eval` here....